test/e2e-go/features/transactions/messageRate_test.go

```diff
@@ -0,0 +1,264 @@
+		time.Sleep(sleepTime)
+	}
+}
+
+func testMessageRateChangesWithTxnRate(t *testing.T, templatePath string, txnRates []uint) {
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	fixture.SetupNoStart(t, templatePath)
+	nodeDataDir, err := fixture.GetNodeDir("Node")
+	a.NoError(err)
+	cfg, err := config.LoadConfigFromDisk(nodeDataDir)
+	a.NoError(err)
+	cfg.EnableVerbosedTransactionSyncLogging = true
+	cfg.SaveToDisk(nodeDataDir)
+	fixture.Start()
+
+	defer fixture.Shutdown()
+
+	client := fixture.GetLibGoalClientForNamedNode("Node")
+	accountsList, err := fixture.GetNodeWalletsSortedByBalance(client.DataDir())
+	a.NoError(err)
+	account := accountsList[0].Address
+	clientAlgod := fixture.GetAlgodClientForController(fixture.GetNodeControllerForDataDir(nodeDataDir))
+
+	// get the node account's secret key
+	secretKey, err := fetchSecretKey(client, nodeDataDir)
+	a.NoError(err)
+	signatureSecrets, err := crypto.SecretKeyToSignatureSecrets(secretKey)
+	a.NoError(err)
+
+	// build the path for the primary node's log file
+	logPath := filepath.Join(nodeDataDir, "node.log")
+
+	// Get the relay's gossip port
+	r1, err := fixture.GetNodeController("Relay1")
+	a.NoError(err)
+	listeningURLRaw, err := r1.GetListeningAddress()
+	a.NoError(err)
+	listeningURL := strings.Split(listeningURLRaw, "//")[1]
+
+	// store message rate for each of the txn rates
+	prevMsgRate := 0.0
+
+	errChan := make(chan error)
+	resetChan := make(chan bool)
+	msgRateChan := make(chan float64)
+	ctx, stopParsing := context.WithCancel(context.Background())
+	defer stopParsing()
+
+	go parseLog(ctx, logPath, listeningURL, errChan, msgRateChan, resetChan)
+
+	for _, txnRate := range txnRates {
+		// get the min transaction fee
+		minTxnFee, _, err := fixture.CurrentMinFeeAndBalance()
+		a.NoError(err)
+		transactionFee := minTxnFee * 1000 * 253
+
+		startTime := time.Now()
+		txnSentCount := uint(0)
+
+		for {
+			// send txns at txnRate for 30s
+			timeSinceStart := time.Since(startTime)
+			if timeSinceStart > 30*time.Second {
+				break
+			}
+
+			tx, err := client.ConstructPayment(account, account, transactionFee, 0, GenerateRandomBytes(8), "", [32]byte{}, 0, 0)
+			a.NoError(err)
+			signedTxn := tx.Sign(signatureSecrets)
+
+			_, err = clientAlgod.SendRawTransaction(signedTxn)
+			a.NoError(err, "Unable to send raw txn")
+
+			txnSentCount++
+
+			throttleTransactionRate(startTime, txnRate, txnSentCount)
+		}
+
+		// wait for some time for the logs to get flushed
+		time.Sleep(2 * time.Second)
+
+		// send reset on resetChan to signal the parseLog goroutine to send the msgRate and reset its counters
+		resetChan <- true
+
+		select {
+		case err := <-errChan:
+			a.Error(err)
+		case msgRate := <-msgRateChan:
+			aErrorMessage := fmt.Sprintf("TxSync message rate not monotonic for txn rate: %d", txnRate)
+			a.GreaterOrEqual(msgRate, prevMsgRate, aErrorMessage)
```

Comment:
Will be helpful to also print the txnRate and msgRate even if the test is passing to have an idea on these numbers.

Reply:
Fixed.